charts/cray-etcd-base/templates/migration-jobs.yaml

```diff
@@ -0,0 +1,252 @@
+
+               cat > /tmp/upload-snapshot.py <<EOF
+               #!/usr/bin/env python3
+
+               import boto3
+               import sys
+
+               def main():
+                   s3 = boto3.resource('s3', endpoint_url=sys.argv[1])
+                   bucket = s3.Bucket('etcd-backup')
+                   file_name = '/tmp/etcd_bitnami_migration.snapshot.db'
+                   cluster="{{ include "cray-etcd-base.fullname" . }}"
+                   key_name="%s/etcd_bitnami_migration.snapshot.db" % (cluster)
+                   bucket.upload_file(Filename=file_name, Key=key_name)
+
+               if __name__ == '__main__':
+                   main()
+               EOF
+               chmod 755 /tmp/upload-snapshot.py
+
+               pods=$(kubectl get endpoints ${cluster}-etcd-client -n ${ns} -o json | jq -r .subsets[].addresses[].targetRef.name)
+               file_name=etcd_bitnami_migration.snapshot.db
+               for pod in $pods
+               do
+                 echo "Taking snapshot from ${pod}..."
+                 kubectl exec -i -n ${ns} ${pod} -c etcd -- /bin/sh -c 'ETCDCTL_API=3 etcdctl snapshot save snapshot.db'
+                 if [ $? -eq 0 ]; then
+                   echo "Copying snapshot from ${pod} to /tmp/${file_name}..."
+                   kubectl -n ${ns} -c etcd cp $pod:snapshot.db /tmp/$file_name
+                   if [ $? -eq 0 ]; then
+                     break
+                   fi
+                 fi
+               done
+
+               echo "Grabbing keys and S3 endpoint from etcd-backup-s3-credentials..."
+               access_key=$(kubectl get secret -n services etcd-backup-s3-credentials -ojsonpath='{.data.access_key}' | base64 -d)
+               secret_key=$(kubectl get secret -n services etcd-backup-s3-credentials -ojsonpath='{.data.secret_key}' | base64 -d)
+               http_s3_endpoint=$(kubectl get secret -n services  etcd-backup-s3-credentials -ojsonpath='{.data.http_s3_endpoint}' | base64 -d)
+
+               mkdir -p /root/.aws
+               cat > /root/.aws/credentials <<EOF
+               [default]
+               aws_access_key_id = $access_key
+               aws_secret_access_key = $secret_key
+               EOF
+
+               cat > /root/.aws/config <<EOF
+               [default]
+               region=foo
+               EOF
+
+               echo "Uploading snapshot ${cluster}/etcd_bitnami_migration.snapshot.db to ${http_s3_endpoint}..."
+               /tmp/upload-snapshot.py ${http_s3_endpoint}
+               echo "Done!"
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "cray-etcd-base.fullname" . }}-etcd-post-install"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
+    helm.sh/hook-delete-policy: before-hook-creation
+spec:
```

Comment:
do you want a ttlSecondsAfterFinished on this job as well?

Reply:
great catch -- yes i do!  will fix..